### PR TITLE
#40 Improve Storybook experience

### LINKED ISSUE #40: Storybook interaction improvements
## Basic information

* Tiller version: 1.0.2
* Module: storybook

## Enhancement description

Enhance controls, code display, viewport sizes and Storybook accessibility. Desirably, also add some guidelines on Storybook usage and interaction. Additionally, update Storybook addons for a more stable experience.

### Current behaviour

Some components have excess controls and lack tokens customization.

### Wanted behaviour

Consistent factories and code display. Easier and clearer Storybook navigation.

### PR DESCRIPTION
**Pull request by fbeceic**
## Basic information

* Tiller version: 1.0.2
* Module: storybook

## Description

Enhance controls, code display, viewport sizes and Storybook accessibility. Desirably, also add some guidelines on Storybook usage and interaction. Additionally, update Storybook addons for a more stable experience.

### Summary

- [x] Update Storybook dependencies to v6.5.16
- [x] Enhance controls used in factories
- [x] Enhance code display for factories
- [x] Provide a usage tutorial for Storybook
- [x] Expand viewport choices when viewing a component

### Related issue

Closes #40 

## Types of changes

- Enhancement (non-breaking change which enhances existing functionality)
- New feature (non-breaking change which adds functionality)
- Docs change
- Dependency upgrade

## Checklist

- [x] I have read the project's **CONTRIBUTING** document
- [x] I have checked my code with the project's static analysis tooling
- [x] I have formatted my code with the project's Prettier code-style configuration
- [x] I have checked my code for misspellings
- [x] I have organized my changes in easy-to-follow commits
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added a story to cover my changes
- [x] All new and existing story tests pass

### CODE REVIEW

File: storybook/src/core/Breadcrumbs.stories.tsx [hunk 1]
```diff
@@ -30,6 +33,14 @@ export default {
   parameters: {
     docs: {
       page: mdx,
+      source: { type: "dynamic", excludeDecorators: true },
+      transformSource: (source) => {
+        const correctedSource = source
+          .replace(/<Breadcrumb>/g, "<Breadcrumbs.Breadcrumb>")
+          .replace(/<\/Breadcrumb>/g, "</Breadcrumbs.Breadcrumb>")
```
Comment (ivam5) [inline storybook/src/core/Breadcrumbs.stories.tsx:40]: /Breadcrumb/g, "Breadcrumbs.Breadcrumb" ?

Comment (fbeceic) [reply storybook/src/core/Breadcrumbs.stories.tsx:40]: If I replace it like that, `<Breadcrumbs...` will also get replaced, so in this case I had to do it like this.

File: storybook/.playroom/frameComponent.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import "../.storybook/index.css";
 export default function FrameComponent({ children }) {
   return (
     <IntlProvider lang="en">
-      <FormikDecorator>
+      <FormikDecorator showFieldValues={false}>
```
Comment (jtomic-croz) [inline storybook/.playroom/frameComponent.js:15]: This is ok for me, I wouldn't remove it

File: storybook/src/alert/NotificationProvider.mdx [hunk 3]
```diff
@@ -20,16 +20,18 @@ a `useNotificationContext()` function imported from our library and assign it to
...
-      **title: "Example Notification",
-      description: "Example notification description."
-      icon: { type: "check-circle", variant: "outline", color: "green" },
-      onDismiss:
-    });
+notifications.push({
+  **title: "Example Notification",
+  description: "Example notification description."
+  icon: { type: "check-circle", variant: "outline", color: "green" },
+  onDismiss:
```
Comment (jtomic-croz) [inline storybook/src/alert/NotificationProvider.mdx:32]: This here misses on dismiss function and won't compile

File: storybook/.playroom/frameComponent.js (hunk 2 above)
Comment (fbeceic) [reply storybook/.playroom/frameComponent.js:15]: Playroom doesn't really provide any functionality and is primarily for prototoyping, that's why I don't see the point of having `{}` beneath the rendered components.

Comment (fbeceic) [reply storybook/.playroom/frameComponent.js:15]: Playroom doesn't really provide support for functionality and is primarily for prototyping, that's why I don't see the point of having `{}` beneath rendered components.

Comment (jtomic-croz) [reply storybook/.playroom/frameComponent.js:15]: Didn't see that its inside playroom. In that case it's ok

**Review by jtomic-croz — APPROVED** [review]

**Pull request merged**